internal/repository/user_email_repository.go

```diff
@@ -0,0 +1,153 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/constants"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/helpers"
+	"go.uber.org/zap"
+)
+
+type EmailStore struct {
+	db *pgxpool.Pool
+}
+
+type EmailStorer interface {
+	GetEmailByProfileID(ctx context.Context, sendRequest SendUserInvitationRequest, tx pgx.Tx) (string, error)
+	GetCreatedByIdByProfileID(ctx context.Context, sendRequest SendUserInvitationRequest, tx pgx.Tx) (int, error)
```

Comment:
Define common function for all requirements 
- define new GetInvitations method and pass the filter.

Reply:
done